### PR TITLE
Schema loading helpers

### PR DESCRIPTION
**Pull request by happz**

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -2163,3 +2165,66 @@ def find_fmf_root(path: str) -> List[str]:
...
+
+
+#
+# JSON schema-based validation helpers
+#
+# Aims at FMF data consumed by TMT, but can be used for any structure.
+#
+
+# `Schema` represents a loaded JSON schema structure. It may be fairly complex,
+# but it's not needed to provide the exhaustive and fully -detailed type since
```
Comment (thrix) [inline tmt/utils.py:2177]:
```suggestion
# but it's not needed to provide the exhaustive and fully detailed type since
```

File: tmt/utils.py [hunk 2]
```diff
@@ -2163,3 +2165,66 @@ def find_fmf_root(path: str) -> List[str]:
...
+    if not os.path.isabs(schema_filepath):
+        schema_filepath = os.path.join(
+            pkg_resources.resource_filename(
+                'tmt', 'schemas'), schema_filepath)
+
+    try:
+        with open(schema_filepath, 'r', encoding='utf-8') as f:
+            return cast(Schema, yaml_to_dict(f.read()))
+
+    except OSError as exc:
```
Comment (thrix) [inline tmt/utils.py:2198]: this seems insufficient, looking at `yaml_to_dict` we should also catch tmt's `GeneralError` and also ruamel's YamlError

File: tmt/utils.py [hunk 3]
```diff
@@ -2163,3 +2165,66 @@ def find_fmf_root(path: str) -> List[str]:
...
+
+    schema_dirpath = pkg_resources.resource_filename('tmt', 'schemas')
+
+    try:
+        for dirpath, dirnames, filenames in os.walk(
+                schema_dirpath, followlinks=True):
+            for filename in filenames:
+                # Ignore all files but YAML files.
+                if os.path.splitext(filename)[
+                        1].lower() not in ('.yaml', '.yml'):
```
Comment (thrix) [inline tmt/utils.py:2220]: We should do something about this autopep8, this so hurts

```suggestion
                extension = os.path.splitext(filename)[1].lower()
                if extension not in ('.yaml', '.yml'):
```

**Review by thrix — CHANGES_REQUESTED** [review]

File: tmt/utils.py (hunk 2 above)
Comment (happz) [reply tmt/utils.py:2579]: Hm, how about catching simply `Exception`, and re-dress all of them as `FileError`?

File: tmt/utils.py (hunk 3 above)
Comment (happz) [reply tmt/utils.py]: Already taken care of, the autopep8 line extension has landed in the meantime.

File: tmt/utils.py (hunk 1 above)
Comment (happz) [reply tmt/utils.py]: Fixed.

File: tmt/utils.py (hunk 2 above)
Comment (thrix) [reply tmt/utils.py:2579]: @happz fine with me ...

Comment (happz) [reply tmt/utils.py]: Fixed.

**Comment by happz** [comment]
Rebased & squashed.

File: tmt/utils.py [hunk 4]
```diff
@@ -2577,3 +2579,65 @@ def find_fmf_root(path: str) -> List[str]:
...
+#
+# JSON schema-based validation helpers
+#
+# Aims at FMF data consumed by TMT, but can be used for any structure.
+#
+
+# `Schema` represents a loaded JSON schema structure. It may be fairly complex,
+# but it's not needed to provide the exhaustive and fully detailed type since
+# TMT code is not actually "reading" it. Loaded schema is passed down to
+# jsonschema library, and while `Any` would be perfectly valid, let's use a
```
Comment (thrix) [inline tmt/utils.py:2593]: @happz seems unfinished sentence with the one below

File: tmt/utils.py [hunk 5]
```diff
@@ -2577,3 +2579,65 @@ def find_fmf_root(path: str) -> List[str]:
...
+
+#
+# JSON schema-based validation helpers
+#
+# Aims at FMF data consumed by TMT, but can be used for any structure.
+#
+
+# `Schema` represents a loaded JSON schema structure. It may be fairly complex,
+# but it's not needed to provide the exhaustive and fully detailed type since
+# TMT code is not actually "reading" it. Loaded schema is passed down to
```
Comment (thrix) [inline tmt/utils.py:2592]: `tmt` pls

File: tmt/utils.py [hunk 6]
```diff
@@ -2577,3 +2579,65 @@ def find_fmf_root(path: str) -> List[str]:
         raise MetadataError(f"No fmf root present inside '{path}'.")
     fmf_roots.sort(key=lambda path: len(path))
     return fmf_roots
+
+
+#
+# JSON schema-based validation helpers
+#
+# Aims at FMF data consumed by TMT, but can be used for any structure.
```
Comment (thrix) [inline tmt/utils.py:2587]: `tmt` please

**Review by thrix — APPROVED** [review]
LGTM, just pls fix the small docs issues I found

File: tmt/utils.py (hunk 4 above)
Comment (happz) [reply tmt/utils.py]: Fixed.

File: tmt/utils.py (hunk 6 above)
Comment (happz) [reply tmt/utils.py]: Fixed.

File: tmt/utils.py (hunk 5 above)
Comment (happz) [reply tmt/utils.py]: Fixed.

**Comment by thrix** [comment]
squashed to 1 commit, preparing for merge

**Review by lukaszachy — APPROVED** [review]

**Pull request merged**